app/controllers/benefits/OtherServicesBenefitsController.scala

```diff
@@ -0,0 +1,109 @@
+    missingInputError = s"benefits.otherServicesBenefits.error.${if (user.isAgent) "agent" else "individual"}"
+  )
+
+  private def redirects(cya: EmploymentCYAModel, taxYear: Int, employmentId: String): Seq[ConditionalRedirect] = {
+    RedirectService.servicesBenefitsRedirects(cya, taxYear, employmentId)
+  }
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getSessionDataResult(taxYear, employmentId) { optCya =>
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya, EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { cya =>
+            cya.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.serviceQuestion)) match {
+            case Some(questionResult) => Future.successful(Ok(otherServicesBenefitsView(yesNoForm.fill(questionResult), taxYear, employmentId)))
+            case None => Future.successful(Ok(otherServicesBenefitsView(yesNoForm, taxYear, employmentId)))
+          }
+        }
+      }
+    }
+  }
+
+  def submit(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getSessionDataResult(taxYear, employmentId) { optCya =>
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya, EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { data =>
+          yesNoForm.bindFromRequest().fold(
+            formWithErrors => Future.successful(BadRequest(otherServicesBenefitsView(formWithErrors, taxYear, employmentId))),
+            yesNo => {
+              val cya = data.employment
+              val benefits = cya.employmentBenefits
+              val utilitiesAndServicesModel = cya.employmentBenefits.flatMap(_.utilitiesAndServicesModel)
+
+              val updatedCyaModel: EmploymentCYAModel = {
+                if (yesNo) {
+                  cya.copy(employmentBenefits = benefits.map(_.copy(utilitiesAndServicesModel =
+                    utilitiesAndServicesModel.map(_.copy(serviceQuestion = Some(true))))))
+                } else {
+                  cya.copy(employmentBenefits = benefits.map(_.copy(utilitiesAndServicesModel =
+                    utilitiesAndServicesModel.map(_.copy(serviceQuestion = Some(false), service = None)))))
+                }
+              }
+
+              employmentSessionService.createOrUpdateSessionData(
+                employmentId,
+                updatedCyaModel,
+                taxYear,
+                data.isPriorSubmission,
+                data.hasPriorBenefits
+              )(errorHandler.internalServerError()) {
+                if (data.isPriorSubmission) {
+                  Redirect(CheckYourBenefitsController.show(taxYear, employmentId))
+                } else {
+                  // TODO: Point to the other services benefits page
```

Comment:
Could you change this todo to make it a little clearer. I'm guessing this should point to the other services benefits amount page?

Reply:
Changed to: // TODO: Point to the other services benefits amount page